src/resolve/resolved-varlink.c

```diff
@@ -519,6 +535,33 @@ static int vl_method_resolve_address(Varlink *link, JsonVariant *parameters, Var
         return 1;
 }
 
+static int vl_method_subscribe_dns_resolves(Varlink *link, JsonVariant *parameters, VarlinkMethodFlags flags, void *userdata) {
+        Manager *m;
+        int r;
+
+        assert(link);
+
+        m = varlink_server_get_userdata(varlink_get_server(link));
+        assert(m);
+
+        if (json_variant_elements(parameters) > 0)
+                return varlink_error_invalid_parameter(link, parameters);
+
+        r = set_ensure_put(&m->varlink_subscription, NULL, varlink_ref(link));
+        if (r < 0) {
+                varlink_unref(link);
+                return log_error_errno(r, "Failed to add subscription to set: %m");
+        }
```

Comment:
Hmm, please increase reference counter on success. instead of decrease on failure...

Reply:
That sounds better, fixed now.